cloudevents/extensions/verifiability.md

```diff
@@ -0,0 +1,443 @@
+# Proposal: Verifiable CloudEvents with DSSE
+
+## Goals
+
+This proposal introduces a design for verifiable CloudEvents that is agnostic of delivery protocols and event formats.
+It allows producers of CloudEvents to sign the events they send, and consumers to cryptographically verify the *authenticity and the integrity* of the events that they receive.
+Through this process consumers can be sure that events were in fact produced by the claimed producer (authenticity), and that the events were received exactly as they were sent, and not modified in transit (integrity)—without needing to trust any intermediaries.
+
+The threats addressed by this proposal are those of malicious actors impersonating CloudEvent producers and of malicious actors modifying messages in transit.
+
+With interoperability in mind, this design opts for simplicity and robustness wherever possible.
+
+## Non-goals
+
+This proposal only applies to individual events.
+It does not give consumers any guarantees about the completeness of the event stream or the order that events are delivered in.
+
+The threats of a malicious actor preventing events from being delivered or swapping their order are not addressed by this proposal.
+Neither are the possibilities of events being accidentally dropped, delivered in the wrong order or the same event being delivered multiple times.
+These challenges can be addressed through means of adding the necessary information inside the event payloads.
+
+Because the CloudEvents specification [requires](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#id) the combination of event `source` and `id` to be unique per event, signature replays for identical events are not considered.
+
+Further, this proposal only aims at *verifiability*.
+It does not aim to enable *confidentiality*.
+Consequently, it does not address the threat of unauthorized parties being able to read CloudEvents that were not meant for them (see [Privacy & Security](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#privacy-and-security) in the CloudEvents spec).
+
+While the design in this proposal *can* be used by authorized intermediaries to modify and re-sign events, it explicitly does not aim to provide a cryptographic audit trail of event modifications.
+
+As a general principle, this proposal aims to avoid cryptographic agility in favor of simplicity.
+
+## Constraints
+
+We have set the following constraints for the proposed design:
+
+**Verifiability MUST be OPTIONAL:** This ensures that the additional burden of producing verification material and performing verification only applies when verifiability is desired, which is not always the case.
```

Comment:
I wonder if it's clear that just because a producer adds the extra verifiable stuff to the CE, the consumer is not required to do anything with it or even know what those extensions mean? Could just be me.